da/celestia/proto/celestia.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+package celestia;
+
+option go_package = "github.com/celestiaorg/optimint/da/celestia/pb";
+import "tendermint/types/types.proto";
+
+service DataRetriever {
+	rpc RetrieveShares (Request) returns (Shares) {}
+	rpc RetrieveStateRelevantMessages (Request) returns (StateRevelantMessages) {}
+	rpc RetrieveMessages (Request) returns (tendermint.types.Messages) {}
+}
+
+message DataAvailabilityHeader {
+	repeated bytes row_roots = 1;
+	repeated bytes column_roots = 2;
+}
+
+message Request {
+	// NamespaceID identifies the namespace in Celestia chain.
+	bytes namespace_id = 1;
+
+	// DataAvailabilityHeader identifies the block in Celestia chain.
+	tendermint.types.DataAvailabilityHeader dah = 2;
+}
+
+message Shares {
+	repeated bytes shares = 1;
+}
```

Comment:
Note that in the case of requesting these via bitswap, the shares would automatically be returned together with their inclusion proof (it's simply how ipfs, or more precisely, ipfs+our plugin works: requesting a leaf results in multiple roundtrips also requesting the inner nodes that proof inclusion of that leaf). 

I assume here the rpc returns this all the share leaves (of a namespace) at once. This must then be including an "application proof" or a namespace proof (e.g. see the [NMT API](https://github.com/celestiaorg/nmt/blob/1d72cffd9fb40ecea93660be2e19ca3c1ecea53f/nmt.go#L135-L188) for that purpose). 

I need to make a test that the proof returned by that API actually matches the hashes that the inner nodes have when the data is requested via bitswap.